fake-vm-component-scripts/fake-vm/launch.py

```diff
@@ -10,25 +10,30 @@
 
 instances = []
 
-def fix_config_keys(source):
-    for k, v in source:
-        yield (k, v) if '.' not in k else (k.split('.')[1], v)
+config_keys = {
+    k: k.split('.')[1] for k in {
+        'configuration.name',
+        'configuration.address',
+        'configuration.login',
+        'configuration.failure',
```

Comment:
Yep, this summary is correct. However, I need to be able to test that the instance name is set correctly through discovery, so some kind of name is still necessary. And since there is no default, the most natural way is to ask it from the user.

Ideally the actual instance name should be used in launch script, and scripts should be able to change instance names of discovered instances (although the name of user-created instances should probably be kept intact regardless of the script output). And vice versa, if the user renames an instance in the portal, it should be passed through to scripts, e.g. as a part of reconfiguration which we will implement soon. We're not there, so it is the closest approximation I could think of.